updater/src/Updater/Generate/Changelog.purs

```diff
@@ -0,0 +1,104 @@
+module Updater.Generate.Changelog where
+
+import Prelude
+
+import Affjax as AX
+import Affjax.ResponseFormat as RF
+import Data.Array (filter, null)
+import Data.Codec (decode)
+import Data.Codec.Argonaut (JsonCodec, array, printJsonDecodeError)
+import Data.Codec.Argonaut as CA
+import Data.Codec.Argonaut.Record as CAR
+import Data.Either (Either(..))
+import Data.Foldable (foldl)
+import Data.Maybe (Maybe(..))
+import Data.Monoid (power)
+import Data.String.CodeUnits (takeWhile)
+import Data.String.Common (joinWith, trim)
+import Data.String.Utils (lines, startsWith)
+import Effect.Aff (Aff, error, throwError)
+import Node.Encoding (Encoding(..))
+import Node.FS.Aff as FSA
+
+type ReleaseInfo =
+  { tag_name :: String
+  , html_url :: String
+  , body :: String
+  , published_at :: String
+  , draft :: Boolean
+  }
+
+releaseCodec :: JsonCodec ReleaseInfo
+releaseCodec =
+  CAR.object "ReleaseInfo" $
+    { tag_name: CA.string
+    , html_url: CA.string
+    , body: CA.string
+    , published_at: CA.string
+    , draft: CA.boolean
+    }
+
+appendReleaseInfoToChangelog :: forall r. { owner :: String, repo :: String | r } -> Aff Unit
+appendReleaseInfoToChangelog gh = do
+  releases <- recursivelyFetchReleases [] 1 gh
+  let
+    realReleases = filter (\r -> r.draft == false) releases
+    appendContent = foldl addReleaseInfo "" realReleases
+  FSA.appendTextFile UTF8 "./CHANGELOG.md" appendContent
+  where
+    addReleaseInfo :: String -> ReleaseInfo -> String
+    addReleaseInfo acc rec =
+      let
+        dateWithoutTimeZone = takeWhile (_ /= 'T') rec.published_at
+        bodyWithFixedHeaders = fixHeaders $ trim rec.body
+      in acc <> joinWith "\n"
+        [ "## [" <> rec.tag_name <> "](" <> rec.html_url <> ") - " <> dateWithoutTimeZone
+        , ""
+        , bodyWithFixedHeaders
+        , ""
+        , ""
+        ]
+
+    fixHeaders :: String -> String
+    fixHeaders s =
+      let
+        incAllHeaderLevels =
+          incHeaderlevel 5
+            >>> incHeaderlevel 4
+            >>> incHeaderlevel 3
+            >>> incHeaderlevel 2
+            >>> incHeaderlevel 1
+      in
+        joinWith "\n" $ map incAllHeaderLevels (lines s)
+
+    incHeaderlevel :: Int -> String -> String
+    incHeaderlevel level line =
+      if startsWith ((power "#" level) <> " ") line
+        then "#" <> line
+        else line
```

Comment:
I'm not totally sure about this one. Ideally none of the releases in -contrib are using markdown headers (ideally we wouldn't have any and just bold any subsections). Maybe we could replace lines that begin with some number of `#` by stripping those and surrounding the result with `**` to bold them?

For example, if a release has a header `# Some Header` then ideally that would become `**Some Header**` rather than `## Some Header`, because that would be at the same level as a release itself.

Would you be open to making that change? Or do you feel that we should retain the headers?

Reply:
I'm not sure whether any of the `-contrib` libraries have headers, but I added this due to testing it on the `purescript/purescript` repo.

Yeah, I can make that change. I think I prefer headers over bolded sections, but I don't think this matters too much. While markdown headers would allow one to use anchor links, it won't matter in this file because there will be multiple headers with the same name, so the anchor links won't work anyway.

Reply:
Done.